src/Disciplina/DB/DSL/SimpleTxDB.hs

```diff
@@ -0,0 +1,155 @@
+import qualified Data.ByteString.Char8 as C
+
+
+-- | Simple transaction database
+newtype SimpleTxDB a = SimpleTxDB
+   { getSimpleTxDB :: Reader [PrivateTx] a
+   } deriving (Functor, Applicative, Monad, MonadReader [PrivateTx])
+
+instance MonadSearchTxObj SimpleTxDB where
+  runTxQuery = runSimpleTxQuery
+  runTxsQuery = runSimpleTxsQuery
+  runObjQuery = undefined -- TODO, implement this one, but how are Object defined?
+
+instance Serialise Address where
+  encode (Address h) = encode h
+  decode = Address <$> decode
+
+-- | TODO, where should we put these?
+instance Serialise Grade
+instance Serialise EducatorTxMsg
+instance Serialise StudentTxMsg
+instance Serialise PrivateTxPayload
+instance Serialise CourseId
+instance Serialise PrivateTx
+
+
+-- | Find Tx in db with hash == h
+runSimpleTxQuery :: QueryTx -> SimpleTxDB (Maybe PrivateTx)
+runSimpleTxQuery (SELECTTx _ (TxIdEq (h :: PrivateTxId))) =
+  safeHead.filter ((==h).hash) <$> ask
+
+-- | Find Txs in db with SubjectId == a
+runSimpleTxsQuery :: QueryTxs -> SimpleTxDB [PrivateTx]
+runSimpleTxsQuery (SELECTTxs _ (TxSubjectIdEq sId)) =
+  filter ((==sId).ciSubject._ptxCourseId) <$> ask
+
+-- | Find Txs in db with grade == g
+runSimpleTxsQuery (SELECTTxs _ (TxGradeEq grade)) =
+  filter ((== Just grade).getGrade._ptxPayload) <$> ask
+  where getGrade (EducatorTx (GradeCourse g)) = Just g
+        getGrade _ = Nothing
+
+-- | Find Txs in db with grade >= g
+runSimpleTxsQuery (SELECTTxs _ (_ :>= grade)) = do
+  filter ((>= Just grade).getGrade._ptxPayload) <$> ask
+  where getGrade (EducatorTx (GradeCourse g)) = Just g
+        getGrade _ = Nothing
+
+runSimpleTxsQuery (SELECTTxs _ (a :& b)) =
+   intersect <$> runQuery (SELECTTxs WHERE a) <*> runQuery (SELECTTxs WHERE b)
+
+runSimpleTxsQuery (SELECTTxs _ (a :|| b)) =
+   union <$> runQuery (SELECTTxs WHERE a) <*> runQuery (SELECTTxs WHERE b)
+
+-- | Find all txs with subjectId which is descendant of sId
+runSimpleTxsQuery (SELECTTxs _ (TxSubjectIsDescendantOf sId)) =
+  filter (isDescendantOf sId . ciSubject ._ptxCourseId) <$> ask
+  where isDescendantOf x y = hasPathFromTo activityTypeGraphIndexed x y
+
+
+-- | run query
+-- given <tx_hash>, I want to be able to get { <tx> | hash(<tx>) = <tx_hash> }
+runFindTx :: PrivateTx -> Maybe PrivateTx
+runFindTx tx = runReader (getSimpleTxDB . runQuery $ q tx) simpleTxDB
+  where q tx = SELECTTx WHERE (TxIdEq (hash tx))
+
+-- | Should return Nothing since mkPrivateTx 'a' 'l'
+-- | does not exist in privateTx db
+testFindTx1 :: Maybe PrivateTx
+testFindTx1 = runFindTx $ mkPrivateTx 'a' 'l'
+
+-- | Should return Just (mkPrivate 'a' 'k')
+testFindTx2 :: Maybe PrivateTx
+testFindTx2 = runFindTx $ mkPrivateTx 'a' 'k'
+
+
+-- | run query
+-- | I want to be able to get { [<tx>] | P(<ordered_tx_param>) }, where P is a system of (in)equalities
+runFindTxs :: QueryTxs -> [PrivateTx]
+runFindTxs q = runReader (getSimpleTxDB . runQuery $ q) simpleTxDB
+
+-- | TODO, move tests to test module
```

Comment:
:heavy_plus_sign: . Everything after `runSimpleTxsQuery` should be in test module.